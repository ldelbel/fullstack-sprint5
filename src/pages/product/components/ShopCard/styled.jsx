import styled from "styled-components";

export const StyledShopCard = styled.div`
  width: 470px;
  height: 255px;
  box-shadow: 0px 5px 5px -3px rgb(0 0 0 / 20%),
    0px 8px 10px 1px rgb(0 0 0 / 14%), 0px 3px 14px 2px rgb(0 0 0 / 12%);
  padding: 2rem 1.5rem;

  > p {
    text-align: center;
    color: #666;
    font-weight: 400;
    margin-bottom: 1.8rem;

    span {
      font-weight: bold;
      color: #4581a8;
      text-decoration: underline;
      cursor: pointer;
    }
  }

  .prices {
    display: flex;
    justify-content: space-between;
    margin-bottom: 1.7rem;

    strong {
      font-size: 1.6rem;
      color: #d0403a;
      display: flex;
      align-items: center;
    }

    > div {
      display: flex;
      flex-direction: column;
      color: #666;
      font-weight: 400;
      font-size: 0.7rem;

      span + span {
        margin-top: 0.4rem;
      }
    }
  }

  .addBtn {
    background: #008844;
    border: none;
    border-radius: 4px;
    width: 100%;
    color: white;
    font-weight: bold;
    font-size: 0.9rem;
    padding: 0.8rem 0;

    &:hover {
      cursor: pointer;
      box-shadow: 0 2px 4px #3b2e2e60;
    }
  }

  .deliver {
    display: flex;
    justify-content: space-between;
    margin-top: 1.5rem;

    &__info {
      display: flex;
      align-items: center;
      height: 2rem;

      span {
        font-size: 0.7rem;
        font-weight: 400;
        color: #333;
        margin-left: 0.5rem;

        &:hover {
          cursor: default;
        }
      }

      input {
        width: 31%;
        font-size: 1rem;
        padding: 0 0.5rem;
        height: 100%;
        border: 1px solid #bbbbbb;
        border-radius: 2px;
        margin-left: 0.5rem;

        &:hover {
          border: 1px solid #585858;
        }
      }

      button {
        height: 100%;
        border: 1px solid #bbbbbb;
        border-radius: 2px;
        background: #fff;
        margin-left: 0.5rem;
        color: #585858;

        &:hover {
          border: 1px solid #585858;
          cursor: pointer;
        }
      }
    }

    p {
      font-size: 0.6rem;
      white-space: nowrap;
      display: flex;
      align-items: center;
      color: #777;
      font-weight: 400;
      text-decoration: underline;

      &:hover {
        cursor: pointer;
        color: #555;
      }
    }
  }
`;
