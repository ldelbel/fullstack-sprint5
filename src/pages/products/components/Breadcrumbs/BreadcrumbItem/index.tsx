import React from 'react';
import { StyledBreadcrumbItem } from "./styled";

interface IBreadcrumbItem {
  link: string;
  name: string;
}

export function BreadcrumbItem({ link, name }: IBreadcrumbItem) {
  return (
    <StyledBreadcrumbItem>
      {link ? (
        <>
          <a className="breadcrumbs__link" href={link}>
            {name}
          </a>
          <span className="breadcrumbs__item  breadcrumbs__separator">/</span>
        </>
      ) : (
        <span className="breadcrumbs__link">{name}</span>
      )}
    </StyledBreadcrumbItem>
  );
}