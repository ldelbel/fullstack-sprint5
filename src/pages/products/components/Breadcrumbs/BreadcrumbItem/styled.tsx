import styled from "styled-components";

export const StyledBreadcrumbItem = styled.div`
  font-size: 0.8125rem;
  font-weight: 600;
  color: #585858;

  @media (max-width: 1200px) {
    display: none;
  }

  &:not(:last-child):hover {
    text-decoration: underline;
  }

  &--active {
    color: #261e1e;
  }

  @media (max-width: 1200px) {
    &--active {
      display: block;
      text-align: center;
      font-size: 1.4375rem;
      font-weight: 400;
      text-transform: uppercase;
    }
  }

  .breadcrumbs__separator {
    margin-left: 8px;
    margin-right: 8px;
    user-select: none;
  }
`;
