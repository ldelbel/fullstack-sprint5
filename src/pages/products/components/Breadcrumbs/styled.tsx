import styled from "styled-components";

export const StyledBreadcrumbs = styled.section`
  margin-bottom: 16px;

  nav {
    max-width: 1200px;
    margin: 0 auto;
  }

  .breadcrumbs__list {
    display: flex;
  }

  @media (max-width: 1200px) {
    .breadcrumbs__list {
      justify-content: center;
    }
  }
`;
