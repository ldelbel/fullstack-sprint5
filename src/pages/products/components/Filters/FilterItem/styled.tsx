import styled from "styled-components";

export const StyledFilterItem = styled.div`
  display: flex;
  align-items: center;
  padding: 12px 14px;

  &:hover {
    cursor: pointer;
    background-color: rgba(38, 30, 30, 0.04);
  }

  .filters__label {
    text-transform: uppercase;
    font-size: 0.6875rem;
    font-weight: 400;
    color: #261e1e;
  }

  .filters__img {
    height: 20px;
    margin-left: 8px;
  }
`;
