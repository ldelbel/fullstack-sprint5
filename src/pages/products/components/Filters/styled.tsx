import styled from "styled-components";

export const StyledFilters = styled.section`
    border-top: 1px solid #e3e3e3;
    border-bottom: 1px solid #e3e3e3;

  @media (max-width: 1200px) {
      display: none;
  }

  .filters__list {
    display: flex;
  }


`;
