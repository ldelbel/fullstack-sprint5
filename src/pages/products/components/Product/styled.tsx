import styled from "styled-components";

export const StyledProduct = styled.div`
  @media (max-width: 1200px) {
    flex-basis: 50%;
  }

  @media (min-width: 1201px) {
    flex-basis: 25%;
  }

  .card {
    cursor: pointer;
    border: 1px solid white;
    padding: 8px;
  }

  .card:hover {
    box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2),
      0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0, 0, 0, 0.12);
  }

  .card__img {
    max-width: 100%;
    max-height: 100%;
  }

  .card__description {
    font-size: 0.6875rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.87);
    text-transform: uppercase;
    text-align: center;
    margin-top: 8px;
    margin-bottom: 10px;
  }

  .card__price {
    font-size: 0.8125rem;
    font-weight: 600;
    text-transform: uppercase;
    text-align: center;
  }
`;
