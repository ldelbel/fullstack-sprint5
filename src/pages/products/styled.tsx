import styled from "styled-components";

export const StyledProductsPage = styled.main`
  margin: 0 auto;
  width: 80%;
  max-width: 1100px;
  padding: 16px;

  .main__filters {
    border-top: 1px solid #e3e3e3;
    border-bottom: 1px solid #e3e3e3;
  }

  @media (max-width: 1200px) {
    .main__filters {
      display: none;
    }
  }

  .main__filters .filters__list {
    display: flex;
  }

  .main__filters .filters__item {
    display: flex;
    align-items: center;
    padding: 12px 14px;
  }

  .main__filters .filters__item:hover {
    cursor: pointer;
    background-color: rgba(38, 30, 30, 0.04);
  }

  .main__filters .filters__label {
    text-transform: uppercase;
    font-size: 0.6875rem;
    font-weight: 400;
    color: #261e1e;
  }

  .main__filters .filters__img {
    height: 20px;
    margin-left: 8px;
  }

  .main__products {
    margin-top: 20px;
    margin-bottom: 20px;
  }

  .main__products .products__row {
    margin: 8px;
  }

  .main__products .products__list {
    display: flex;
    justify-content: space-between;
  }

  @media (max-width: 1200px) {
    .main__products .products__list {
      flex-wrap: wrap;
    }
  }

  @media (min-width: 1201px) {
    .main__products .products__list {
      flex-wrap: wrap;
    }
  }
`;
