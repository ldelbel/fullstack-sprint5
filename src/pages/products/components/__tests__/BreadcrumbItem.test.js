import React from "react";
import { render } from "@testing-library/react";
import { BreadcrumbItem } from "../Breadcrumbs/BreadcrumbItem/index";
import renderer from "react-test-renderer";
import { StyledBreadcrumbItem } from '../Breadcrumbs/BreadcrumbItem/styled';

describe("BreadcrumbItem component", () => {
  
  it("should render breadcrumbItem", () => {
    const mockStyledComponent = () => <StyledBreadcrumbItem />
    jest.mock('../Breadcrumbs/BreadcrumbItem/styled', () => mockStyledComponent() )
    const { getByText, getByRole } = render(
      <BreadcrumbItem link="linktest" name="nametest" />
    );
    expect(getByRole('link').hasAttribute('href')).toBeTruthy();
    expect(getByText("nametest")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const breadcrumb = renderer.create(<BreadcrumbItem link={"link-test"} name={"name-test"} />);
    expect(breadcrumb).toMatchSnapshot();
  });

});
