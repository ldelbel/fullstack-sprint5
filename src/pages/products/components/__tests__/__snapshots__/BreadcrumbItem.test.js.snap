// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbItem component should match snapshot 1`] = `
<div
  className="sc-bdnxRM dhyORQ"
>
  <a
    className="breadcrumbs__link"
    href="link-test"
  >
    name-test
  </a>
  <span
    className="breadcrumbs__item  breadcrumbs__separator"
  >
    /
  </span>
</div>
`;
